src/main/java/org/jabref/model/pdf/FileAnnotation.java

```diff
@@ -1,127 +1,116 @@
 package org.jabref.model.pdf;
 
 
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Optional;
+
 import org.apache.pdfbox.cos.COSDictionary;
 import org.apache.pdfbox.cos.COSName;
 import org.apache.pdfbox.pdmodel.interactive.annotation.PDAnnotation;
 
-public class FileAnnotation {
+public final class FileAnnotation {
 
-    private String commentId;
-    private String author;
-    private String date;
-    private int page;
-    private String content;
-    private String annotationType;
-    private FileAnnotation linkedFileAnnotation;
     private final static int ABBREVIATED_ANNOTATION_NAME_LENGTH = 45;
+    private final static String ANNOTATION_DATE_REGEX = "D:\\d+";
+    public final String author;
+    public final String date;
+    public final int page;
+    public final String content;
+    public final String annotationType;
+    public final Optional<FileAnnotation> linkedFileAnnotation;
 
-    private boolean linkedAnnotation;
 
-    public FileAnnotation(final String commentId, final String author, final String date, final int page,
+    /**
+     * A flexible constructor, mainly used as dummy if there is actually no annotation.
+     *
+     * @param author         The authors of the annotation
+     * @param date           The last modified date of the annotation
+     * @param pageNumber     The page of the pdf where the annotation occurs
+     * @param content        the actual content of the annotation
+     * @param annotationType the type of the annotation
+     */
+    public FileAnnotation(final String author, final String date, final int pageNumber,
                           final String content, final String annotationType) {
         this.author = author;
-        this.date = date;
-        this.page = page;
+        this.date = prettyPrint(date);
+        this.page = pageNumber;
         this.content = content;
-        this.commentId = commentId;
         this.annotationType = annotationType;
+        this.linkedFileAnnotation = Optional.empty();
     }
 
-    public FileAnnotation(final PDAnnotation annotation, final int page ){
+    /**
+     * Creating a normal FileAnnotation from a PDAnnotation.
+     *
+     * @param annotation The actual annotation that holds the information
+     * @param pageNumber The page of the pdf where the annotation occurs
+     */
+    public FileAnnotation(final PDAnnotation annotation, final int pageNumber) {
         COSDictionary dict = annotation.getDictionary();
```

Comment:
Can you please reuse the "dummy" construction, i.e. `this(annotation.getModifiedDate(), pageNumber, ...)`. This comment also applies to the second constructor below.